From 08bdbc6ef46ad522a24dc6b8e01c039cb0c7e761 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Tue, 27 Jul 2021 09:30:29 +0900
Subject: [PATCH] ksmbd: use channel signingkey for binding SMB2 session setup
Git-commit: 08bdbc6ef46ad522a24dc6b8e01c039cb0c7e761
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Windows client disconnect connection by wrong signed SMB2 session
setup response on SMB3 multichannel mode.

Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/ksmbd/smb2pdu.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs/ksmbd/smb2pdu.c b/fs/ksmbd/smb2pdu.c
index 93e98ef94c73..7f5ca471b306 100644
--- a/fs/ksmbd/smb2pdu.c
+++ b/fs/ksmbd/smb2pdu.c
@@ -8150,7 +8150,8 @@ void smb3_set_sign_rsp(struct ksmbd_work *work)
 		len = ALIGN(len, 8);
 	}
 
-	if (le16_to_cpu(hdr->Command) == SMB2_SESSION_SETUP_HE) {
+	if (conn->binding == false &&
+	    le16_to_cpu(hdr->Command) == SMB2_SESSION_SETUP_HE) {
 		signing_key = work->sess->smb3signingkey;
 	} else {
 		chann = lookup_chann_list(work->sess, work->conn);
-- 
2.33.0

